### PR TITLE
[bitnami/kafka] Append zookeeperChrootPath when using external zookeeper.

### LINKED ISSUE #9989: [bitnami/kafka] `zookeeperChrootPath` is ignored in when using `externalZookeeper.servers`.
### Name and Version

bitnami/kafka 3.1.0

### What steps will reproduce the bug?

1. Deploy bitnami/zookeeper as standalone chart with 3 replicas.
2. Deploy bitnami/kafka using the standalone zookeeper pods as externalZookeeper, and specify a zookeeperChrootPath
3. Result: the zookeeperChrootPath is ignored, and kafka creates nodes in the zookeeper root /

### Are you using any custom parameters or values?
```yaml
zookeeper:
  enabled: false
zookeeperChrootPath: "/kafka"
externalZookeeper:
  servers:
  - zookeeper-0.zookeeper-headless
  - zookeeper-1.zookeeper-headless
  - zookeeper-2.zookeeper-headless
```
### What is the expected behavior?

All kafka data for zookeeper should be written in the /kafka sub-folder

```
I have no name!@zookeeper-1:/$ zkCli.sh ls -R /kafka
/opt/bitnami/java/bin/java
Connecting to localhost:2181

WATCHER::

WatchedEvent state:SyncConnected type:None path:null
/kafka
/kafka/admin
/kafka/brokers
/kafka/cluster
/kafka/config
/kafka/consumers
/kafka/isr_change_notification
/kafka/latest_producer_id_block
/kafka/log_dir_event_notification
/kafka/admin/delete_topics
/kafka/brokers/ids
/kafka/brokers/seqid
/kafka/brokers/topics
/kafka/cluster/id
/kafka/config/brokers
/kafka/config/changes
/kafka/config/clients
/kafka/config/ips
/kafka/config/topics
/kafka/config/users

```

### What do you see instead?

```
I have no name!@zookeeper-1:/$ zkCli.sh ls -R /
/opt/bitnami/java/bin/java
Connecting to localhost:2181

WATCHER::

WatchedEvent state:SyncConnected type:None path:null
/admin
/brokers
/cluster
/config
/consumers
...
```

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by geertp-ngdata**
### Description of the change

Append zookeeperChrootPath when externalZookeeper.servers are used.

### Benefits

When using an external zookeeper, it's typically because we share it with other applications using zookeeper, so it's 
important to use a root path where kafka keeps its data. 
There is a documented zookeeperChrootPath value, but it was ignored when using external zookeepers.

### Applicable issues

  - fixes #9989 

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/kafka/templates/statefulset.yaml [hunk 1]
```diff
@@ -181,7 +181,7 @@ spec:
               {{- if .Values.zookeeper.enabled }}
               value: {{ printf "%s%s" (include "kafka.zookeeper.fullname" .) (tpl .Values.zookeeperChrootPath .) | quote }}
               {{- else }}
-              value: {{ include "common.tplvalues.render" (dict "value" (join "," .Values.externalZookeeper.servers) "context" $) }}
+              value: {{ include "common.tplvalues.render" (dict "value" (printf "%s%s" (join "," .Values.externalZookeeper.servers) (tpl .Values.zookeeperChrootPath .)) "context" $)    }}
```
Comment (migruiz4) [inline bitnami/kafka/templates/statefulset.yaml:184]: With the current approach, `zookeeperChrootPath` will only be applied to the last Zookeeper server:
```
            - name: KAFKA_CFG_ZOOKEEPER_CONNECT
              value: test1,test2,test3/path
```

When the objective should be:
```
            - name: KAFKA_CFG_ZOOKEEPER_CONNECT
              value: test1/path,test2/path,test3/path
```

To get this result, I suggest using
```suggestion
              {{- $zookeeperExternal := list }}
              {{- range .Values.externalZookeeper.servers }}
              {{- $zookeeperExternal = append $zookeeperExternal (printf "%s%s" (include "common.tplvalues.render" ( dict "value" . "context" $)) (tpl $.Values.zookeeperChrootPath $)) }}
              {{- end }}
              value: {{ join "," $zookeeperExternal | quote }}
```

**Review by migruiz4 — CHANGES_REQUESTED** [review]
Hi @geertp-ngdata,

Thank you very much for your contribution! Please take a look at my comment.

File: bitnami/kafka/templates/statefulset.yaml (hunk 1 above)
Comment (geertp-ngdata) [reply bitnami/kafka/templates/statefulset.yaml:184]: @migruiz4 I tried your approach first, that results in:

`KAFKA_CFG_ZOOKEEPER_CONNECT: zookeeper-0.zookeeper-headless/kafka,zookeeper-1.zookeeper-headless/kafka,zookeeper-2.zookeeper-headless/kafka`

But that's not how the configuration works. It seems that kafka first splits at the first `/`, and then interpretes everything before the '/' as the server quorum, and everything after the first `/` as the path.

I had that with the current version when I tried:

```yaml
externalZookeeper:
  servers:
  - zookeeper-0.zookeeper-headless/kafka
  - zookeeper-1.zookeeper-headless/kafka
  - zookeeper-2.zookeeper-headless/kafka
```

This gives weird results:

```
Server(s): zookeeper-0.zookeeper-headless
Path:
  /kafka,zookeeper-1.zookeeper-headless
    /kafka,zookeeper-2.zookeeper-headless
      /kafka
        /admin
```

In my approach, the result is correct:

```
Server(s): zookeeper-0.zookeeper-headless,zookeeper-1.zookeeper-headless,zookeeper-2.zookeeper-headless
Path:  /kafka
```

Comment (migruiz4) [reply bitnami/kafka/templates/statefulset.yaml:184]: My bad!

I thought the syntax was: `host:port/path,host:port/path`.

Indeed, the correct syntax is `host:port,host:port,host:port/path`.

For reference: https://docs.confluent.io/platform/current/installation/configuration/broker-configs.html#brokerconfigs_zookeeper.connect

**Review by migruiz4 — APPROVED** [review]

**Pull request merged**